src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -484,6 +503,17 @@ mono_threads_wasm_on_thread_attached (void)
 #endif
 }
 
+void
+mono_threads_wasm_on_managed_thread_started (MonoNativeThreadId pthread_id, gboolean external_eventloop, gboolean threadpool_thread)
+{
+#ifdef DISABLE_THREADS
+	return;
```

Comment:
I can imagine this (the on_managed_thread_started) not running in a non-threads build could be a bad surprise for someone later who tries to add logic to the `mono_wasm_pthread_on_managed_thread_started` function, maybe make sure there's a comment there which points out it won't run in single-threaded builds

Reply:
I added comments to the relevant TS method. I also added condition to skip `mono_wasm_pthread_on_managed_thread_started` on the main thread. In `mono-threads-wasm.c` there is already lot of `DISABLE_THREADS` noise giving hints about it.